### PR TITLE
fix (saml): code refactoring

### PR DESCRIPTION
**Pull request by jimlambrt**
It may be easier to review this PR by commit.

### CODE REVIEW

File: saml/models/core/response_test.go [hunk 1]
```diff
@@ -46,7 +46,6 @@ func Test_ParseResponse_ResponseContainer(t *testing.T) {
 
 	r.Equal(res.Destination, "http://localhost:8000/saml/acs")
 	r.Equal(res.ID, "saml-response-id")
-	r.Equal(res.InResponseTo, "saml-request-id")
```
Comment (jimlambrt) [inline saml/models/core/response_test.go:49]: There doesn't appear to be `res.InResponseTo`

File: saml/models/core/response_test.go [hunk 2]
```diff
@@ -71,14 +71,6 @@ var responseXMLStatus = `<?xml version="1.0" encoding="UTF-8"?>
    </saml2p:Status>
 </saml2p:Response>`
 
-func Test_ParseResponse_Status(t *testing.T) {
```
Comment (jimlambrt) [inline saml/models/core/response_test.go:74]: there doesn't appear to be a `Status` field.

File: saml/response_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+// TODO: add the ability to sign requests, so we can write more complete unit tests
```
Comment (jimlambrt) [inline saml/response_test.go:14]: This TODO is important, so we can write more complete unit tests.

File: saml/models/core/response_test.go (hunk 2 above)
Comment (hcjulz) [reply saml/models/core/response_test.go:74]: I wonder why. It was def. defined once. Will add it back, but thx for fixing this. I can add tests back once we have the status field added again.

**Review by hcjulz — APPROVED** [review]
👏 thx @jimlambrt LGTM

**Review by austingebauer — APPROVED** [review]
Thanks, Jim!

File: saml/response.go [hunk 4]
```diff
@@ -110,55 +123,57 @@ func (sp *ServiceProvider) ParseResponse(
...
+				return nil, fmt.Errorf("%s: %w", op, ErrMissingAttributeStmt)
 			}
 		}
 	}
 
 	// TODO: transform gosaml2 response to core.Response
 	var result core.Response
-	xml.Unmarshal([]byte(samlResp), &result)
-
+	if xml.Unmarshal([]byte(samlResp), &result); err != nil {
```
[the remark this answers is not in the document]
Comment (austingebauer) [reply saml/response.go:143]: Yes, thanks for the reminder. I'll put the PR up for this. Lots of changes on the Vault side recently and lost track 😅

**Pull request merged**